src/utils.rs

```diff
@@ -852,11 +852,12 @@ pub fn setting_or_default(req: &Request<Body>, name: &str, default: String) -> S
 // Detect and redirect in the event of a random subreddit
 pub async fn catch_random(sub: &str, additional: &str) -> Result<Response<Body>, String> {
 	if sub == "random" || sub == "randnsfw" {
-		let new_sub = json(format!("/r/{sub}/about.json?raw_json=1"), false).await?["data"]["display_name"]
-			.as_str()
-			.unwrap_or_default()
-			.to_string();
-		Ok(redirect(&format!("/r/{new_sub}{additional}")))
+		Ok(redirect(&format!(
+			"/r/{}{additional}",
+			json(format!("/r/{sub}/about.json?raw_json=1"), false).await?["data"]["display_name"]
+				.as_str()
+				.unwrap_or_default()
+		)))
```

Comment:
Instead of removing the intermediate variable, just keep it for ease of readability and understanding, and remove the string call.

Reply:
Ooh... never mind, I see. Hate when that happens. I think it's worth it to avoid the string call, even if it's rare, the readability benefit is tiny